### PR TITLE
chore: add documentation for local node module

### LINKED ISSUE #645: Document/replace local node module, postman-templating
**Is your feature request related to a problem? Please describe.**
Working with the local node module, postman-templating, is an anti pattern and it is quite hard to find information on how to make changes to it.

Struggled for few hours to debug deployment to staging.

**Describe the solution you'd like**
Clear guide on how to update, install dependencies. IE: Needs to delete package-lock.json in parent module in order for it to install the correct node modules on postman-templating.

**Describe alternatives you've considered**
Move the local node module into a separate github repo and npm install from there, but it comes with a different set of problems and challenges.

### PR DESCRIPTION
**Pull request by lshaowei18**
## Problem

It was unclear how to make changes to the local node module, `postman-templating`.

Closes #645 

@chinying do let me know if I missed out some parts or there are some things are still unclear.

### CODE REVIEW

File: docs/local-module.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+## Local development
+
+To test the changes you want to make to a module like `postman-templating`:
+
+1. Make changes to the module.
+2. Run `npm start build` in `postman-templating`.
+
+`npm start build` will remove & replace the `build` folder in `postman-templating` with the updated change.
+
```
Comment (chinying) [inline docs/local-module.md:39]: This part is technically correct, but just doesn't paint the full picture. 
Eg. If I were to be testing my changes made to `postman-templating` in the `/frontend` directory, I will also have to run `npm install` or `npm ci` **under the `frontend` directory** for the new `postman-templating` module to be built there.

File: docs/local-module.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+- [Background](#background)
+- [How to install](#how-to-install)
+- [Local development](#local-development)
+- [Build dependencies](#build-dependencies)
+- [More information](#more-information)
+
+## Background
+
+We wanted to find a way to share the templating logic because:
```
Comment (chinying) [inline docs/local-module.md:11]: I think it's a bit strange for someone new to the repo to suddenly see this sentence without context. Perhaps start by giving a brief explanation of what a local module is. Sharing the templating logic is just an example of a local module.

**Review by chinying — CHANGES_REQUESTED** [review]

File: docs/local-module.md (hunk 1 above)
Comment (lshaowei18) [reply docs/local-module.md:39]: Hmmm i tested on the backend and it seems like i didn't have to `npm install` or `npm ci`, let me verify it.

**Comment by lshaowei18** [comment]
Added a new section on deploying & tried to give more background to what a local node module is, hopefully that makes things more clear.

File: docs/local-module.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+To test the changes you want to make to a module like `postman-templating`:
+
+1. Make changes to the module.
+2. Run `npm start build` in `postman-templating`.
+
+`npm start build` will remove & replace the `build` folder in `postman-templating` with the updated change.
+
+## Deploying or building in Travis
+
+If there are any new dependencies installed in the local node module, the root folder's `package-lock.json` needs to be updated. This can be done through deleting the `node_modules` and running `npm install`.
```
Comment (chinying) [inline docs/local-module.md:46]: Change "root" to "parent" for clarity. Root implies the top level.

**Review by chinying — CHANGES_REQUESTED** [review]

File: docs/local-module.md (hunk 3 above)
Comment (lshaowei18) [reply docs/local-module.md:46]: Good point!

**Review by chinying — APPROVED** [review]

**Pull request merged**